plugin/trino-mongodb/src/test/java/io/trino/plugin/mongodb/TestMetadataPrivilegesRequired.java

```diff
@@ -0,0 +1,173 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.plugin.mongodb;
+
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+import com.mongodb.client.MongoClient;
+import com.mongodb.client.MongoClients;
+import com.mongodb.client.MongoDatabase;
+import io.trino.testing.DistributedQueryRunner;
+import io.trino.testing.QueryRunner;
+import io.trino.util.AutoCloseableCloser;
+import org.bson.BsonArray;
+import org.bson.BsonDocument;
+import org.bson.BsonString;
+import org.testng.annotations.AfterClass;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Test;
+
+import java.util.Locale;
+import java.util.Map;
+import java.util.Optional;
+
+import static io.airlift.testing.Closeables.closeAllSuppress;
+import static io.trino.testing.TestingSession.testSessionBuilder;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+
+public class TestMetadataPrivilegesRequired
+{
+    private static final String ROOT_USER = "root";
+    private static final String ROOT_PASSWORD = "password";
+    private static final String TEST_USER = "testUser";
+    private static final String TEST_PASSWORD = "pass";
+    private static final String TEST_ROLE = "testRole";
+    private static final String TEST_DATABASE = "test";
+    private static final String TEST_COLLECTION = "testCollection";
+
+    private final AutoCloseableCloser afterClassCloser = AutoCloseableCloser.create();
+
+    private QueryRunner queryRunnerWithDefaultSettings;
+    private QueryRunner queryRunner;
+
+    @BeforeClass
+    public void setUp()
+            throws Exception
+    {
+        MongoServer mongo = afterClassCloser.register(setupMongoServer());
+        Map<String, String> defaultMongoProperties = ImmutableMap.of(
+                "mongodb.case-insensitive-name-matching", "true",
+                "mongodb.connection-url", mongo.getConnectionString(TEST_USER, TEST_PASSWORD).toString() + "/?authSource=" + TEST_DATABASE);
+        queryRunnerWithDefaultSettings = afterClassCloser.register(createQueryRunner(defaultMongoProperties));
+        queryRunner = afterClassCloser.register(createQueryRunner(ImmutableMap.<String, String>builder()
+                .putAll(defaultMongoProperties)
+                .put("mongodb.metadata-privileges-required", "false")
+                .buildOrThrow()));
+    }
+
+    @AfterClass(alwaysRun = true)
+    public void tearDown()
+            throws Exception
+    {
+        try (afterClassCloser) {
+            //
+        }
+        finally {
+            queryRunnerWithDefaultSettings = null;
```

Comment:
Either `AutoClosable` or `finally` clause.
Move those 2 statements to setup method as a:
``` 
afterClassCloser.register(()->{
    queryRunnerWithDefaultSettings = null;
    queryRunner = null;
});
```
and just run `afterClassCloser.close()` here

Reply:
Can we make `TestMetadataPrivilegesRequired` to extend  `AbstractTestQueryFramework` ?